docs/testplan/OSPF-test-plan.md

```diff
@@ -0,0 +1,85 @@
+    - [Test Case #1 \- Test Dynamic Routing Change](#test-case-1---test-dynamic-routing-change)
+        - [Test Objective](#test-objective-1)
+        - [Test Steps](#test-steps-1)
+
+## Overview
+The purpose of this test is to verify the functionality of OSPF routing sessions on a SONIC switch DUT. This test aims to cover the routing functionality of both OSPF and OSPFv3.
+The test assumes all standard topology configurations, such as BGP neighborship, are pre-configured in the DUT and neighboring systems with no OSPF configurations. It is also assumed that neighboring devices are all SONiC devices.
+
+### Scope
+The test is targeting a running SONIC system with fully functioning configuration.
+The purpose of the test is functional testing of OSPF on a SONIC system, testing OSPF neighborship status and dynamic routing changes in the event of a link failure.
+
+### Testbed
+The test is avilable for all variations of **Tier 0** and **Tier 1** topologies available in the **sonic-mgmt** repository where the neighboring devices are SONiC devices.
+
+### Related DUT CLI Commands
+Existing BGP routing functionality is disabled using the following command inside the configure terminal of SONiC FRR (vtysh) mode.
+
+    sonic(config)# no router bgp
+
+OSPF configurations are be made using the following commands inside the configure terminal of SONiC FRR (vtysh) mode.
+
+    #Enter OSPF configuration mode
+    sonic(config)# router ospf
+
+    #Configure OSPF neighbor network
+    sonic(config-router)# network <IP-Address>/<Netmask> area <Area>
+
+
+## Test Structure
+### Setup Configuration
+The test assumes all standard configurations, such as BGP neighborship, are pre-configured in the DUT and neighboring systems with no OSPF configurations. It is also assumed that the neighboring devices are of SONiC type.
+
+### Testbed Setup
+
+#### Tier 0 (t0)
+![Variation t0](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/img/testbed-t0.png?raw=true)
+
+- The DUT has 32 ports.
+- Requires 4 VMs.
+- The first 28 ports are connected to PTF docker simulating servers.
+- The last 4 ports are connected to 4 VMs simulating T1 devices. The connection to each of the upstream T1 is configured as a port-channel with single link.
+
+#### Tier 1 (t1)
+![Variation t1](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/img/testbed-t1.png?raw=true)
+
+- The DUT has 32 ports.
+- Requires 32 VMs.
+- 16 of the ports are connected to 16 VMs simulating upstream T2 neighbors. The connections to upstream T2s are single link without port channel configured.
+- 16 of the ports are connected to another 16 VMs simulating downstream T0 neighbors. No port-channel is configured for the links between DUT and T0 neighbors.
+
+## Test Cases
+### Test Case \#0 - Test Neighborship Status
+#### Test Objective
+Verify whether OSPF neighbors are successfully created and appear with active status.
+
+#### Test Steps
+1. Establish OSPF sessions and neighbors
```

Comment:
Suggest make the OSPF configuration a plugin that can be extended to cEOS neighbors.

Make OSPF configuration and tear down plugin invokable from outside the test. and it should be smart enough to detect the config is already in OSPF therefore skip config/teardown.

Making config/teardown independent enough so that it can be used as a test-suite in the future: e.g. config once, run the test in this plan, run other existing IO tests to verify data plane, then tear down.

Reply:
Incorporated in test plan as per suggestion. We will implement the plugins in our test case code.